samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -144,6 +141,83 @@ class DailyCallManager {
     }
   }
 
+  /**
+   * Handles participant-left event:
+   * - Cleans up the video and audio tracks for the participant
+   * - Removes the related UI elements
+   * @param {Object} event - The participant-* event object.
+   */
+  handleParticipantLeft(event) {
+    const participantId = event.participant.session_id;
+
+    // Clean up the video and audio tracks for the participant
+    this.destroyTracks(['video', 'audio'], participantId);
+
+    // Now, remove the related UI elements
+    const videoContainer = document.getElementById(
+      `video-container-${participantId}`
+    );
+    if (videoContainer) {
+      videoContainer.remove();
+    }
+  }
+
+  /**
+   * Handles participant-joined and participant-updated events:
+   * - Updates the participant count
+   * - Updates device states for the local participant
+   * - Manages video and audio tracks based on their current state
+   * @param {Object} event - The participant-* event object.
+   */
+  handleParticipantJoinedOrUpdated(event) {
+    const { participant } = event;
+    const participantId = participant.session_id;
+    const isLocal = participant.local;
+    const tracks = participant.tracks;
+
+    // Always update the participant count regardless of the event action
+    this.updateAndDisplayParticipantCount();
+
+    Object.entries(tracks).forEach(([trackType, trackInfo]) => {
+      // Update the video UI based on the track's state
+      if (trackType === 'video') {
+        this.updateVideoUi(trackInfo, participantId);
+      }
+
+      // Update the camera and microphone states for the local user based on the track's state
+      if (isLocal) {
+        this.updateUiForDevicesState(trackType, trackInfo);
+      }
+
+      // If a track exists...
+      if (trackInfo.persistentTrack) {
+        if (trackType === 'video') {
+          // Attach and play the video
+          this.playVideo(trackInfo, participantId);
```

Comment:
i think this should be more of a `startOrUpdateVideo()` function, that gracefully handles the cases:
- no video element existed before
- a video element already existed, and
  - `persistentTrack` hasn't changed from before (avoid creating a new `srcObject`)
  - `persistentTrack` has changed from before

Reply:
same feedback for `playAudio()`. avoiding creating a new `srcObject` and `play()`ing it whenever possible is especially important for audio since i think `play()` will fail in background tabs in safari.

Reply:
Makes sense. This is actually the source of a bug that I just noticed. When muting and unmuting the mic, the video flickers. Thanks for the tip!